game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1805,25 +1805,16 @@ private void defenderRetreatSubs(final IDelegateBridge bridge) {
    * Check for suicide units and kill them immediately (they get to shoot back, which is the point).
    */
   private void checkSuicideUnits(final IDelegateBridge bridge) {
-    if (isDefendingSuicideAndMunitionUnitsDoNotFire()) {
-      final List<Unit> deadUnits =
-          CollectionUtils.getMatches(attackingUnits, Matches.unitIsSuicide());
-      bridge
-          .getDisplayChannelBroadcaster()
-          .deadUnitNotification(battleId, attacker, deadUnits, dependentUnits);
-      remove(deadUnits, bridge, battleSite, false);
-    } else {
-      final List<Unit> deadUnits = new ArrayList<>();
-      deadUnits.addAll(CollectionUtils.getMatches(defendingUnits, Matches.unitIsSuicide()));
-      deadUnits.addAll(CollectionUtils.getMatches(attackingUnits, Matches.unitIsSuicide()));
-      bridge
-          .getDisplayChannelBroadcaster()
-          .deadUnitNotification(battleId, attacker, deadUnits, dependentUnits);
-      bridge
-          .getDisplayChannelBroadcaster()
-          .deadUnitNotification(battleId, defender, deadUnits, dependentUnits);
-      remove(deadUnits, bridge, battleSite, null);
-    }
+    final List<Unit> deadUnits = new ArrayList<>();
+    deadUnits.addAll(CollectionUtils.getMatches(attackingUnits, Matches.unitIsSuicideOnAttack()));
+    deadUnits.addAll(CollectionUtils.getMatches(defendingUnits, Matches.unitIsSuicideOnDefense()));
+    bridge
```

Comment:
This deadUnit notification only appears in one branch of the unmodified code, for a case of `isDefendingSuicideAndMunitionUnitsDoNotFire`, why should this be shown now? Is this a no-op perhaps, is there something subtle that is not obvious?

Reply:
I think you are looking at outdated code here.